platform/ios/src/MGLScaleBar.mm

```diff
@@ -0,0 +1,384 @@
+#import <Mapbox/Mapbox.h>
+#import "MGLScaleBar.h"
+
+static const CGFloat MGLFeetPerMile = 5280;
+
+struct MGLRow {
+    CLLocationDistance distance;
+    NSUInteger numberOfBars;
+};
+
+static const struct {
+    struct MGLRow row;
```

Comment:
This is one extra level of indirection. All we need is an array of MGLRows; this creates an array of anonymous structs containing MGLRows. With a simple array of MGLRows, we can eliminate some of the braces below.

My earlier suggestion was that, as long as you don't need to refer to the specific type of these rows anywhere else, it's possible to inline MGLRow here, without naming it, and use `auto` everywhere else you'd refer to the row's type.